async_hooks: executionAsyncResource matches in hooks

Flarna opened the pull request:
Ensure that resource returned by `executionAsyncResource()` in before and after hook matches that resource causing this before/after calls.

##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

Refs: #30959
fyi @Qard

Flarna commented:
Seems the test I added here fails in node-test-commit-custom-suites-freestyle. Will take a look later. Any hints whats spacial about this CI job are welcome.

addaleax commented:
@flarna Not sure why the job is named that way but it’s basically running each test inside a Worker thread, i.e. `python tools/test.py --worker path/to/test`

vdeturckheim commented:
@Flarna the issue seems fixed, is it ready for a rebase?

Flarna commented:
From my point of view it's ready for landing.

vdeturckheim commented:
@Flarna can you please squash it to a single commit?

Flarna commented:
Sure, but why is this needed here?
Till now this happened during landing. At least I was never asked before to squash on private branch in node repo.

vdeturckheim commented:
It's indeed doable at merge time but I don't like rewriting other people's commits 😅

Flarna commented:
Ok, rebased and squashed. Not sure if CI needs to run again now.

I was assuming that squashin is done by some landing script or similar. Commits are rewritten anyway during landing as PR url, list of revierers is added.

vdeturckheim commented:
Thanks @Flarna , I'll restart CI.
Yep commits are updated, but it is automatically done by a script.

vdeturckheim commented:
Landed in fd3d02ac3266
